### PR TITLE
Remove Confirmation from plaid bank flow

### PR DESCRIPTION
**Pull request by parasharrajat**
### Fixed Issues

$ https://github.com/Expensify/App/issues/48730
$ https://github.com/Expensify/App/issues/48732
PROPOSAL:

### Tests
Same as QA steps.

- [ ] Verify that no errors appear in the JS console

### Offline tests

1. Sign up with a Gmail account added to Workspace feeds beta
2. Enable Expensify Card in more features
3. Select Issue card to kick off the enablement flow.
3. Start setup using credentials from option 3 in the [SO](https://stackoverflowteams.com/c/expensify/questions/342).

	 ##### For Plaid flow. 

 		This Flow can not be completed while offline. 

	 ##### For manual Flow. 

 		a. After you have added routing and account number, click next. 
 		b. Verify that the confirmation screen is not shown again on step 1. 

5. Finish the bank account flow. 
6. Go through the issue card flow, and land on the card name step. 
7. verify that the cad name field is not prefilled with a suggestion. 

### QA Steps
1. Sign up with a Gmail account added to Workspace feeds beta
2. Enable Expensify Card in more features
3. Select Issue card to kick off the enablement flow.
3. Start setup using credentials from option 3 in the [SO](https://stackoverflowteams.com/c/expensify/questions/342).

	 ##### For Plaid flow. 

 		a. After plaid flow is completed, select the account on the Step 1. 
		b. Verify that clicking next does not reconfim the selected account. 

	 ##### For manual Flow. 

 		a. After you have added routing and account number, click next. 
 		b. Verify that the confirmation screen is not shown again on step 1. 

5. Finish the bank account flow. 
6. Go through the issue card flow, and land on the card name step. 
7. verify that the cad name field is not prefilled with a suggestion. 

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/2a350512-53fa-4348-8134-d6f39a23e349

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/7dfe4d59-4c9d-4b24-aabc-79352ee3af92

https://github.com/user-attachments/assets/0ca21462-e394-42b1-8ea5-8fd3fca5a3e8

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

**Comment by trjExpensify** [comment]
>could you raise this in slack from Vit, I am not sure about this one.

Did this get escalated to slack? CC: @mountiny

File: src/pages/ReimbursementAccount/BankInfo/substeps/Plaid.tsx [hunk 1]
```diff
@@ -36,24 +36,26 @@ function Plaid({reimbursementAccount, reimbursementAccountDraft, onNext, plaidDa
     const isFocused = useIsFocused();
     const selectedPlaidAccountID = reimbursementAccountDraft?.[BANK_INFO_STEP_KEYS.PLAID_ACCOUNT_ID] ?? '';
 
-    const handleNextPress = useCallback(() => {
-        const selectedPlaidBankAccount = (plaidData?.bankAccounts ?? []).find(
-            (account) => account.plaidAccountID === reimbursementAccountDraft?.[BANK_INFO_STEP_KEYS.PLAID_ACCOUNT_ID] ?? null,
-        );
+    const selectPlaidAccount = useCallback(
+        (plaidAccountID: string) => {
+            ReimbursementAccountActions.updateReimbursementAccountDraft({plaidAccountID});
```
Comment (koko57) [inline src/pages/ReimbursementAccount/BankInfo/substeps/Plaid.tsx:41]: @parasharrajat why do you want to update plaidAccountID first?

**Comment by trjExpensify** [comment]
Are we going to be able to have this reviewed and merged today?

**Comment by trjExpensify** [comment]
This seems to have stalled in review. Can we get an update, please?

**Comment by mountiny** [comment]
Yeah I have bumped Rajat in Slack [here](https://expensify.slack.com/archives/C049HHMV9SM/p1726300890807989?thread_ts=1725872659.033869&cid=C049HHMV9SM) on Saturday

File: src/pages/workspace/expensifyCard/issueNew/CardNameStep.tsx [hunk 2]
```diff
@@ -26,10 +25,6 @@ function CardNameStep() {
     const [issueNewCard] = useOnyx(ONYXKEYS.ISSUE_NEW_EXPENSIFY_CARD);
 
     const isEditing = issueNewCard?.isEditing;
-    const data = issueNewCard?.data;
-
-    const userName = PersonalDetailsUtils.getUserNameByEmail(data?.assigneeEmail ?? '', 'firstName');
-    const defaultCardTitle = `${userName}'s Card`;
```
Comment (koko57) [inline src/pages/workspace/expensifyCard/issueNew/CardNameStep.tsx:32]: @parasharrajat why did you remove this logic?

File: src/pages/ReimbursementAccount/BankInfo/substeps/Manual.tsx [hunk 3]
```diff
@@ -61,7 +61,7 @@ function Manual({reimbursementAccount, onNext}: ManualProps) {
         [translate],
     );
 
-    const shouldDisableInputs = !!(reimbursementAccount?.achData?.bankAccountID ?? '');
+    const hasbankAccountData = !!(reimbursementAccount?.achData?.bankAccountID ?? '');
```
Comment (koko57) [inline src/pages/ReimbursementAccount/BankInfo/substeps/Manual.tsx:64]:
```suggestion
    const hasBankAccountData = !!(reimbursementAccount?.achData?.bankAccountID ?? '');
```
Capitalize Bank

File: src/pages/ReimbursementAccount/BankInfo/substeps/Manual.tsx [hunk 4]
```diff
@@ -89,9 +89,8 @@ function Manual({reimbursementAccount, onNext}: ManualProps) {
                 role={CONST.ROLE.PRESENTATION}
                 defaultValue={defaultValues[BANK_INFO_STEP_KEYS.ROUTING_NUMBER]}
                 inputMode={CONST.INPUT_MODE.NUMERIC}
-                disabled={shouldDisableInputs}
                 shouldSaveDraft
-                shouldUseDefaultValue={shouldDisableInputs}
+                shouldUseDefaultValue={hasbankAccountData}
```
Comment (koko57) [inline src/pages/ReimbursementAccount/BankInfo/substeps/Manual.tsx:93]:
```suggestion
                shouldUseDefaultValue={hasBankAccountData}
```
as above

File: src/pages/ReimbursementAccount/BankInfo/substeps/Manual.tsx [hunk 5]
```diff
@@ -102,9 +101,8 @@ function Manual({reimbursementAccount, onNext}: ManualProps) {
                 role={CONST.ROLE.PRESENTATION}
                 defaultValue={defaultValues[BANK_INFO_STEP_KEYS.ACCOUNT_NUMBER]}
                 inputMode={CONST.INPUT_MODE.NUMERIC}
-                disabled={shouldDisableInputs}
                 shouldSaveDraft
-                shouldUseDefaultValue={shouldDisableInputs}
+                shouldUseDefaultValue={hasbankAccountData}
```
Comment (koko57) [inline src/pages/ReimbursementAccount/BankInfo/substeps/Manual.tsx:105]:
```suggestion
                shouldUseDefaultValue={hasBankAccountData}
```
same 😃

File: src/pages/workspace/expensifyCard/issueNew/CardNameStep.tsx (hunk 2 above)
Comment (koko57) [reply src/pages/workspace/expensifyCard/issueNew/CardNameStep.tsx:32]: https://github.com/Expensify/App/issues/48732 I know this issue, but shouldn't we have the default name for the physical card?

Comment (koko57) [reply src/pages/workspace/expensifyCard/issueNew/CardNameStep.tsx:32]: yes, it's for both physical and virtual

Comment (koko57) [reply src/pages/workspace/expensifyCard/issueNew/CardNameStep.tsx:32]: you should check for `issueNewCard?.data?.cardType` if it's CONST.EXPENSIFY_CARD.CARD_TYPE.PHYSICAL should be "[userName]'s Card" like it's implemented now

am I right @mountiny?

Comment (mountiny) [reply src/pages/workspace/expensifyCard/issueNew/CardNameStep.tsx:32]: Yeah only physical cards should have the suggested name per the GH

**Review by ishpaul777 — APPROVED** [review]

**Comment by mountiny** [comment]
> I am stuck with the account verification step. it is asking for 3 transaction values.

Did you put in `credentials_good` in the plaid modal for the mobile verification?

**Comment by mountiny** [comment]
@parasharrajat, can you merge the main? The patch was not applied correctly in the test but I can see the action succeed on other PRs

File: src/pages/ReimbursementAccount/BankInfo/BankInfo.tsx [hunk 6]
```diff
@@ -61,37 +50,41 @@ function BankInfo({reimbursementAccount, reimbursementAccountDraft, plaidLinkTok
...
-                    [BANK_INFO_STEP_KEYS.PLAID_ACCESS_TOKEN]: values[BANK_INFO_STEP_KEYS.PLAID_ACCESS_TOKEN] ?? '',
-                    [BANK_INFO_STEP_KEYS.PLAID_MASK]: values[BANK_INFO_STEP_KEYS.PLAID_MASK] ?? '',
-                    [BANK_INFO_STEP_KEYS.IS_SAVINGS]: values[BANK_INFO_STEP_KEYS.IS_SAVINGS] ?? false,
-                },
-                policyID,
-            );
-        }
-    }, [setupType, values, bankAccountID, policyID]);
+    const submit = useCallback(
+        (submitData: unknown) => {
```
Comment (mountiny) [inline src/pages/ReimbursementAccount/BankInfo/BankInfo.tsx:54]: Any way we could improve this type? @koko57 @blazejkustra @fabioh8010

**Review by mountiny — APPROVED** [review]
Changes look good to me have you tested this after the useOnyx migration @ishpaul777 @parasharrajat?

**Comment by mountiny** [comment]
It was quite a lot of changes so would be great to confirm that it works well 

Thanks for working on this @parasharrajat @ishpaul777

**Review by ishpaul777 — APPROVED** [review]
LGTM!

**Review by mountiny — APPROVED** [review]
Thanks everyone!

**Comment by m-natarajan** [comment]
@parasharrajat Can you explain the step 1

> Sign up with a Gmail account added to Workspace feeds beta

**Comment by mountiny** [comment]
@m-natarajan the beta is no longer relevant, you can use any account and you should be able to enable the cards there
